### ISSUE TITLE
Separate callbacks for on_process_result

### ISSUE THREAD

**Issue by Makcal** [Tishka17/aiogram_dialog#429]
I have a dialog that can start several different dialogs, but is limited to have one `on_process_result()` callback. This is why to distinguish these calls I pass `start_data` in form of `{"intent": "enter_name", "input": ...}` and my callback looks like:
```python
match start_data["intent"]:
    case "enter_name":
        # process input of the name field
        ...
    case "enter_text":
        # process input of the text field
        ...
```

This can produce long and complex callbacks. I wonder if there can be a cleaner way to process results. Thinking a while I've come up with two proposals:

1. Pass a callback directly in a dialog's call:
```python
await manager.start(PromptSG.main, promt_params, callback=process_name)`
```
2. Use a decorator to mark a function as a callback for a particular dialog:
```python
@dialog.callback_for(PromptDialog)
def process_name(start_data, result, manager)
    ...
```

**Comment by Tishka17** [comment]
In version 2.2 you can set on process result callback on window. This is not the same, but anyway, it can reduce size of function.

**Comment by Makcal** [comment]
Yeah, that's not the same, however it would be great to have clear separation for a window, e.g. for buttons that start dialogs

**Comment by dantetemplar** [comment]
upvote+
=
